common/beerocks/bwl/shared/nl80211_client_impl.cpp

```diff
@@ -17,13 +21,276 @@
+                            ctx.width_160 = true;
+                            break;
+                        }
+                    } else {
+                        LOG(DEBUG) << "NL80211_BAND_ATTR_VHT_CAPA attribute is missing";
+                    }
+
+                    if (tb_band[NL80211_BAND_ATTR_VHT_MCS_SET]) {
+                        size_t expected_length = sizeof(band.vht_mcs_set);
+                        size_t actual_length   = nla_len(tb_band[NL80211_BAND_ATTR_VHT_MCS_SET]);
+
+                        if (actual_length <= expected_length) {
+                            memcpy(band.vht_mcs_set,
+                                   nla_data(tb_band[NL80211_BAND_ATTR_VHT_MCS_SET]), actual_length);
+                        } else {
+                            LOG(DEBUG) << "Invalid length for NL80211_BAND_ATTR_VHT_MCS_SET "
+                                          "attribute. Expected length: "
+                                       << expected_length << ", actual length: " << actual_length;
+                        }
+                    } else {
+                        LOG(DEBUG) << "NL80211_BAND_ATTR_VHT_MCS_SET attribute is missing";
+                    }
+
+                    if (tb_band[NL80211_BAND_ATTR_FREQS]) {
+                        struct nlattr *nl_freq;
+                        int rem_freq;
+                        uint16_t max_bandwidth_MHz = 0;
+                        static struct nla_policy freq_policy[NL80211_FREQUENCY_ATTR_MAX + 1];
+                        freq_policy[NL80211_FREQUENCY_ATTR_FREQ]         = {NLA_U32, 0, 0};
+                        freq_policy[NL80211_FREQUENCY_ATTR_DISABLED]     = {NLA_FLAG, 0, 0};
+                        freq_policy[NL80211_FREQUENCY_ATTR_RADAR]        = {NLA_FLAG, 0, 0};
+                        freq_policy[NL80211_FREQUENCY_ATTR_MAX_TX_POWER] = {NLA_U32, 0, 0};
+
+                        nla_for_each_nested(nl_freq, tb_band[NL80211_BAND_ATTR_FREQS], rem_freq)
+                        {
+                            struct nlattr *tb_freq[NL80211_FREQUENCY_ATTR_MAX + 1];
+                            sChannelInfo channel{};
+
+                            if (nla_parse(tb_freq, NL80211_FREQUENCY_ATTR_MAX,
+                                          static_cast<nlattr *>(nla_data(nl_freq)),
+                                          nla_len(nl_freq), freq_policy)) {
+                                LOG(ERROR)
+                                    << "Failed to parse frequency " << nl_freq->nla_type + 1 << "!";
+                                return false;
+                            }
+
+                            if (!tb_freq[NL80211_FREQUENCY_ATTR_FREQ] ||
+                                tb_freq[NL80211_FREQUENCY_ATTR_DISABLED] ||
+                                (tb_freq[NL80211_FREQUENCY_ATTR_NO_IR] &&
+                                 tb_freq[__NL80211_FREQUENCY_ATTR_NO_IBSS]) ||
+                                tb_freq[__NL80211_FREQUENCY_ATTR_NO_IBSS]) {
```

Comment:
`(a && b) || b` is equivalent to just `b`. In other words:
```suggestion
                                tb_freq[__NL80211_FREQUENCY_ATTR_NO_IBSS]) {
```

But maybe I missed something.

Reply:
Fixed